src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -975,19 +976,23 @@ void WriteLog<I>::aio_read_data_blocks(
     [this, log_entries, bls, ctx](int r) {
       for (unsigned int i = 0; i < log_entries.size(); i++) {
         bufferlist valid_data_bl;
-        auto length = log_entries[i]->is_write() ? log_entries[i]->write_bytes :
-                                                   log_entries[i]->ws_datalen;
+	//auto entry = (static_pointer_cast<WriteLogEntry>(log_entries[i]));
+	std::shared_ptr<WriteLogEntry> entry = static_pointer_cast<WriteLogEntry>(log_entries[i]);
+
+        auto length = (entry->ram_entry).is_write() ? (entry->ram_entry).write_bytes
+						      : (entry->ram_entry).ws_datalen;
```

Comment:
Are parenthesis around `entry->ram_entry` necessary?

Reply:
no, it can be removed.  it have been removed.